### DISCUSSION TITLE
How can get Prisma types?

### DISCUSSION THREAD

**Discussion by MariuzM** [prisma/prisma#4841] [Q&A] [answered]
I'm cashing my Prisma connection instance `PrismaClient` but because I'm exporting the instance like so I'm unable to get Prisma type when clicking CTRL + Space
![Screen Shot 2020-12-31 at 15 19 20](https://user-images.githubusercontent.com/36937039/103415776-90f57c80-4b7b-11eb-8a4f-8ce2dfafa012.png)

What types i need to get so i can assign to that `cashedClient` variable? 
```js
import { PrismaClient } from '@prisma/client'

let cachedClient: any

export const prisma = async () => {
  cachedClient ? console.info('Cached DB') : console.info('Not Cached DB')
  if (cachedClient) return { client: cachedClient }
  const client = new PrismaClient()
  cachedClient = client
  return { client }
}

```

**Comment by pantharshit00** [comment, marked as answer]
Just type it as `PrismaClient`.

![image](https://user-images.githubusercontent.com/22195362/103415961-771f5e00-4baa-11eb-8cf6-4dde02230a56.png)

**Reply by MariuzM** [reply]
Hm it does work in the main file where the Client is but not after i have exported `client` to be used in diff file 
![Screen Shot 2020-12-31 at 15 29 55](https://user-images.githubusercontent.com/36937039/103416112-0b72cc00-4b7d-11eb-9af6-93a32da330dc.png)

![Screen Shot 2020-12-31 at 15 27 44](https://user-images.githubusercontent.com/36937039/103416065-c51d6d00-4b7c-11eb-8576-84f4e939561f.png)

**Reply by MariuzM** [reply]
Sorry looks like I'm doing something wrong here i think i figured out

**Reply by MariuzM** [reply]
Looks like doing distructuring strips types 

`const client = await prisma()`
vs 
`const { client } = await prisma()` - stripped types

**Reply by pantharshit00** [reply]
Explicity type the return type of the `prisma` function for that:
```ts
let cachedClient: PrismaClient;

export const prisma = async (): Promise<{ client: PrismaClient }> => {
  cachedClient ? console.info("Cached DB") : console.info("Not Cached DB");
  if (cachedClient) return { client: cachedClient };
  const client = new PrismaClient();
  cachedClient = client;
  return { client };
};
```

<img width="877" alt="image" src="https://user-images.githubusercontent.com/22195362/103438271-9e803480-4c57-11eb-8c7d-26dc0de4933d.png">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
